pkg/controller/operators/catalog/operator.go

```diff
@@ -1654,18 +1686,70 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 				plan.Status.Plan[i].Status = status
 
 			default:
-				return v1alpha1.ErrInvalidInstallPlan
-			}
+				if !isSupported(step.Resource.Kind) {
+					// Not a supported resource
+					plan.Status.Plan[i].Status = v1alpha1.StepStatusUnsupportedResource
+					return v1alpha1.ErrInvalidInstallPlan
+				}
+
+				// Entering unstructured territory... To infinity and beyond!
```

Comment:
This line is unnecessary and should be removed unless you can add something more informative such "Attempt to install other supported resources". You don't really install unstructured resources, more like installing supported resources by decoding yaml into unstructured object first.

Reply:
Will remove